pkg/util/validation/errors.go

```diff
@@ -155,29 +183,40 @@ func (e *exemplarValidationError) Error() string {
 	return fmt.Sprintf(e.message, e.timestamp, mimirpb.FromLabelAdaptersToLabels(e.seriesLabels).String(), mimirpb.FromLabelAdaptersToLabels(e.exemplarLabels).String())
 }
 
-func newExemplarEmtpyLabelsError(seriesLabels []mimirpb.LabelAdapter, exemplarLabels []mimirpb.LabelAdapter, timestamp int64) ValidationError {
+var exemplarEmptyLabelsMsgFormat = globalerrors.Format(
+	globalerrors.ErrIDExemplarLabelsMissing,
+	"received exemplar with no valid labels, timestamp: %%d series: %%s labels: %%s")
+
+func newExemplarEmptyLabelsError(seriesLabels []mimirpb.LabelAdapter, exemplarLabels []mimirpb.LabelAdapter, timestamp int64) ValidationError {
 	return &exemplarValidationError{
-		message:        "exemplar missing labels, timestamp: %d series: %s labels: %s",
+		message:        exemplarEmptyLabelsMsgFormat,
 		seriesLabels:   seriesLabels,
 		exemplarLabels: exemplarLabels,
 		timestamp:      timestamp,
 	}
 }
 
+var exemplarMissingTimestampMsgFormat = globalerrors.Format(
+	globalerrors.ErrIDExemplarTimestampInvalid,
+	"received an exemplar with no timestamp, timestamp: %%d series: %%s labels: %%s")
+
 func newExemplarMissingTimestampError(seriesLabels []mimirpb.LabelAdapter, exemplarLabels []mimirpb.LabelAdapter, timestamp int64) ValidationError {
 	return &exemplarValidationError{
-		message:        "exemplar missing timestamp, timestamp: %d series: %s labels: %s",
+		message:        exemplarMissingTimestampMsgFormat,
 		seriesLabels:   seriesLabels,
 		exemplarLabels: exemplarLabels,
 		timestamp:      timestamp,
 	}
 }
 
-var labelLenMsg = "exemplar combined labelset exceeds " + strconv.Itoa(ExemplarMaxLabelSetLength) + " characters, timestamp: %d series: %s labels: %s"
+var exemplarMaxLabelLengthMsgFormat = globalerrors.Format(
+	globalerrors.ErrIDExemplarLabelsTooLong,
+	"received exemplar whose combined labels set size exceeds the limit of %d characters, timestamp: %%d series: %%s labels: %%s",
+	ExemplarMaxLabelSetLength)
```

Comment:
Nitpick: I think we don't have anything written about this so feel free to ignore it.

I'd drop this (and the rest) of closing parenthesis to a new line. Go has trailing commas to allow consistency, and when the call has its first arg dropped to a new line, the consistency is to have the closing parenthesis on it's own line (and with previous indentation too).

i.e. I'd say:

```golang
// OK: every arg is indented on the same line and ends with a comma <3
var foo = bar.Baz(
	one,
	two,
)


// OK because the opening parenthesis is not alone.
logger.Log("my very long message",
	arg1, arg2)

// NOT OK: looks like an automated formatting tool did this.
var foo = bar.Baz(
	one,
	two)


// NOT OK: an exaggerated version of the previous one, we wouldn't do this, right?
var foo = bar.Baz(
	one)
```

Reply:
Typically I don't want to deal with this kind of stuff. Either there's a linter (and possible autofix tool) or I don't care. I value my time more than where a parenthesis should be put.